sdk/core/azure-core/inc/azure/core/etag.hpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define ETag.
+ */
+
+#pragma once
+
+#include "azure/core/nullable.hpp"
+
+#include <string>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Represents an HTTP validator.
```

Comment:
validator?

```suggestion
   * @brief Represents an HTTP entity tag.
```

Reply:
An ETag is our implementation of the validator used by HTTP conditional requests.

Reply:
Isn't validators a bit overloaded and describe different concepts:
- the comparison mechanisms (strong vs weak)
- the generic term describing any version to compare against, including the last-modified date
 
ETags are a specific type of validator, and they use two types of validation.

How about this:
```suggestion
   * @brief Represents an entity tag which is an opaque string, uniquely identifying each version, and used an HTTP validator.
```